src/Tribe/Widget/Widget_Interface.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace Tribe\Widget;
+
+use Tribe\Events\Views\V2\View_Interface;
+
+/**
+ * Interface Widget_Interface
+ *
+ * @since   TBD
+ *@package Tribe\Widget
+ *
+ */
+interface Widget_Interface {
+
+	/**
+	 * Constructor for V2 Widgets.
+	 *
+	 * @since TBD
+	 *
+	 * @param string              $id_base         Optional. Base ID for the widget, lowercase and unique. If left empty,
+	 *                                             a portion of the widget's class name will be used. Has to be unique.
+	 * @param string              $name            Name for the widget displayed on the configuration page.
+	 * @param array<string,mixed> $widget_options  Optional. Widget options. See wp_register_sidebar_widget() for
+	 *                                             information on accepted arguments. Default empty array.
+	 * @param array<string,mixed> $control_options Optional. Widget control options. See wp_register_widget_control() for
+	 *                                             information on accepted arguments. Default empty array.
+	 */
+	public function __construct( $id_base = '', $name = '', $widget_options = [], $control_options = [] );
+
+	/**
+	 * Echoes the widget content.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $args     Display arguments including 'before_title', 'after_title',
+	 *                                      'before_widget', and 'after_widget'.
+	 * @param array<string,mixed> $instance The settings for the particular instance of the widget.
+	 */
+	public function widget( $args, $instance );
+
+	/**
+	 * Updates a particular instance of a widget.
+	 *
+	 * This function should check that `$new_instance` is set correctly. The newly-calculated
+	 * value of `$instance` should be returned. If false is returned, the instance won't be
+	 * saved/updated.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $new_instance New settings for this instance as input by the user via
+	 *                                          WP_Widget::form().
+	 * @param array<string,mixed> $old_instance Old settings for this instance.
+	 *
+	 * @return array<string,mixed> Settings to save or bool false to cancel saving.
+	 */
+	public function update( $new_instance, $old_instance );
+
+	/**
+	 * Outputs the settings update form.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $instance Current settings.
+	 *
+	 * @return string Default return is 'noform'.
+	 */
+	public function form( $instance );
+
+	/**
+	 * Returns the widget slug that allows the widget to be built via the widget class by slug.
```

Comment:
The preciseness here is a bit confusing - is there more than one widget slug? If so maybe we should use different terms.

Reply:
```suggestion
	 * Returns the widget slug that allows the widget to be built via the widget class using that slug.
```